MoltenVK/MoltenVK/GPUObjects/MVKPipeline.mm

```diff
@@ -337,42 +337,54 @@
 
 static const char vtxCompilerType[] = "Vertex stage pipeline for tessellation";
 
-id<MTLComputePipelineState> MVKGraphicsPipeline::getTessVertexStageState() {
-    MTLComputePipelineDescriptor* plDesc = [_mtlTessVertexStageDesc copy];  // temp retain a copy to be thread-safe.
-    plDesc.computeFunction = _mtlTessVertexFunctions[0];
-    id<MTLComputePipelineState> plState = getOrCompilePipeline(plDesc, _mtlTessVertexStageState, vtxCompilerType);
-    [plDesc release];                                                       // temp release
-    return plState;
-}
-
-id<MTLComputePipelineState> MVKGraphicsPipeline::getTessVertexStageIndex16State() {
-    MTLComputePipelineDescriptor* plDesc = [_mtlTessVertexStageDesc copy];  // temp retain a copy to be thread-safe.
-    plDesc.computeFunction = _mtlTessVertexFunctions[1];
-    plDesc.stageInputDescriptor.indexType = MTLIndexTypeUInt16;
+bool MVKGraphicsPipeline::compileTessVertexStageState(MTLComputePipelineDescriptor* vtxPLDesc,
+													  id<MTLFunction>* vtxFunctions,
+													  VkPipelineCreationFeedback* pVertexFB) {
+	uint64_t startTime = 0;
+    if (pVertexFB) {
+		startTime = mvkGetTimestamp();
+	}
+    vtxPLDesc.computeFunction = vtxFunctions[0];
+    bool res = !!getOrCompilePipeline(vtxPLDesc, _mtlTessVertexStageState, vtxCompilerType);
+
+    vtxPLDesc.computeFunction = vtxFunctions[1];
+    vtxPLDesc.stageInputDescriptor.indexType = MTLIndexTypeUInt16;
     for (uint32_t i = 0; i < 31; i++) {
-		MTLBufferLayoutDescriptor* blDesc = plDesc.stageInputDescriptor.layouts[i];
+		MTLBufferLayoutDescriptor* blDesc = vtxPLDesc.stageInputDescriptor.layouts[i];
         if (blDesc.stepFunction == MTLStepFunctionThreadPositionInGridX) {
                 blDesc.stepFunction = MTLStepFunctionThreadPositionInGridXIndexed;
         }
     }
-    id<MTLComputePipelineState> plState = getOrCompilePipeline(plDesc, _mtlTessVertexStageIndex16State, vtxCompilerType);
-    [plDesc release];                                                       // temp release
-    return plState;
+    res |= !!getOrCompilePipeline(vtxPLDesc, _mtlTessVertexStageIndex16State, vtxCompilerType);
+
+    vtxPLDesc.computeFunction = vtxFunctions[2];
+    vtxPLDesc.stageInputDescriptor.indexType = MTLIndexTypeUInt32;
+    res |= !!getOrCompilePipeline(vtxPLDesc, _mtlTessVertexStageIndex32State, vtxCompilerType);
+
+	if (pVertexFB) {
+		if (!res) {
+			// Compilation of the shader will have enabled the flag, so I need to turn it off.
+			mvkDisableFlags(pVertexFB->flags, VK_PIPELINE_CREATION_FEEDBACK_VALID_BIT);
+		}
+		pVertexFB->duration += (mvkGetTimestamp() - startTime) * mvkGetTimestampPeriod();
```

Comment:
`(mvkGetTimestamp() - startTime) * mvkGetTimestampPeriod()` appears a lot as copy-paste. 

Can we move this to `MVKOSExtensions.h/mm` as a function similar to `mvkGetElapsedMilliseconds()`? 

For example:

```
uint64_t mvkGetElapsedTime(uint64_t startTimestamp = 0, uint64_t endTimestamp = 0);

uint64_t mvkGetElapsedTime(uint64_t startTimestamp, uint64_t endTimestamp) {
	if (endTimestamp == 0) { endTimestamp = mvkGetTimestamp(); }
	return (endTimestamp - startTimestamp) * _mvkTimestampPeriod;
}
```